woocommerce/woocommerce
# Initialize BlockTemplatesController for block themes only

## What?
Closes https://github.com/woocommerce/woocommerce/issues/46463
Supersedes https://github.com/woocommerce/woocommerce/pull/48385

- Move the `wc_blocks_use_blockified_product_grid_block_as_template` option check from BTC to `wc-template-functions.php` as it's shared between classic and block themes.
- Effectively switch the entire BTC class to be only initialized for block themes. 

## How to test

For the `wc_blocks_use_blockified_product_grid_block_as_template` option part, ensure:
- it's switched on when switching from classic to block theme,
- stays switched on when switching from block to block theme,
- is switched off when switching to a classic theme,
- stays switched off when switching from classic to classic theme.

For the BTC initialization switch part:

- Generally, if the tests pass, it should be good to go. The only risk here is that we missed some functional leaks into classic themes, so we might want to do some manual smoke testing. Therefore, switch a classic theme on (e.g. Storefront) and ensure you can edit and view the content as expected.
